docs/06-contributors/999-rfcs/2022-04-30-preflight-instance-id.md

```diff
@@ -0,0 +1,70 @@
+---
+title: (Draft) Unique Preflight Instances ID
+description: How to create unique IDs for preflight instances
+---
+- **Author(s):**: @ekeren
+- **Submission Date**: {2022-04-30}
+- **Stage**: Draft
+
+## Background
+
+In the current implementation of Wing, when instantiating a new preflight class, the compiler provides a default ID for the instance. 
+In cases where the default ID cannot be made unique, a compiler error occurs.
+
+In the last wingathon, more than 70% of preflight instances in the Wing research repository are created with a custom ID. 
+
+## Observed Patterns
+
+When examining the code, the following patterns can be observed:
+
+1. Improved readability of preflight instances, e.g.:
+  
+    [dogfooding/publications-etl](https://github.com/winglang/research/blob/926453f5c5abd2e3158b4c8c659f1e790c99d741//app.w#L70):
+     ```ts (wing)
+     this.initialized = new cloud.Counter() as "initialized";
+     ```
+2. Name duplications - situations where the ID (or part of it) is also reflected in the constructor:
```

Comment:
For this use case, I am wondering if the resource has a physical name (in this case `name: "slack-token"`), then maybe the resource class itself can suggest an identifier that's based on that physical name?